nf_core/pipeline-template/workflows/pipeline.nf

```diff
@@ -82,7 +82,7 @@ workflow {{ short_name|upper }} {
     ch_versions = ch_versions.mix(FASTQC.out.versions.first())
 
     CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
+        ch_versions.unique{ it.text }.collectFile(name: 'collated_versions.yml')
```

Comment:
This change appears to have introduced a bug, a lot of people's pipelines are hanging after the latest pipeline merge. See discussion in Slack [here](https://nfcore.slack.com/archives/CE5LG7WMB/p1671006803103609).

I guess that it was to make it more similar to other instances such as:
```
versions = versions.mix(BISMARK.out.versions.unique{ it.baseName })
```

However, we're not in a `mix()` here and it seems to silently hang forever without any errors. I think we need to revert this change unless anyone has any objections? I'll make an issue..

Reply:
I don't really understand why the template CI checks didn't pick up on this. I found it because the methylseq CI failed due to timeouts. I would have expected the tools CI tests to do the same 🤔

Reply:
This is a mixed channel (a few lines above). It's surprising that checking the text for uniqueness is causing a hang. There would appear to be a deeper cause, but I'm on leave now, so I don't have time to dig into it. It's definitely strange though. If it fixes the issue then revert it, but need to make sure duplicate versions.yml are not kept

Reply:
Actually, perhaps check if the dumpsoftware versions script is also removing duplicate entries by loading them into a dictionary. Then the `unique` is not needed.

Reply:
@ewels OK. I figured out why `unique{ it.text }` is causing a hang.
People are doing this in their workflows:
```
ch_versions.mix(TOOL.out.versions.first().ifEmpty(null))
```
specifically, people are putting `null` objects in their versions channels. They don't need to emit anything ...

The correct code should be
```
ch_versions.mix(TOOL.out.versions.first())
```
Mixing an empty channel is fine.
My guess is there are null pointer exceptions under the hood, resulting in the hang. 

That's why this was never caught before being integrated into workflows.

Reply:
Ok, nice detective work! Sorry to bother you whilst you're away.

Is there any functional difference between `unique()` and `unique{ it.text }`?

Reply:
> Actually, perhaps check if the dumpsoftware versions script is also removing duplicate entries by loading them into a dictionary. Then the unique is not needed.

I have a feeling that even if the Python code is doing this, having duplicate keys within the YAML might make it invalid and crash the loading process? (not tested)

Reply:
> Is there any functional difference between `unique()` and `unique{ it.text }`?

Yes, but I don't know how often the issue of duplication occurs. `unique()` is only checking for uniquely named files ( I think it must be checking the full path which includes the work directory ), which means here it's actually doing nothing, since the work directories are always unique. While most people use `first()` in their workflows, it doesn't help removing occurrences of modules used under different aliases for example from subworkflows. For example, `versions` from, `WORKFLOW_A:SAMTOOLS` and `WORKFLOW_B:SAMTOOLS`, should both be present in the versions channel, and since their work directories are unique, `unique()` won't do anything, but `unique{ it.text }` should remove the second occurrence.